pyteal/ast/subroutine.py

```diff
@@ -610,12 +611,77 @@ def method_signature(self, overriding_name: str = None) -> str:
         return f"{overriding_name}({','.join(args)}){self.type_of()}"
 
     def method_spec(self) -> sdk_abi.Method:
-        skip_names = ["return", "output"]
+        skip_names = ["return", self.OUTPUT_ARG_NAME]
+
+        desc: str = ""
+        arg_descs: dict[str, str] = {}
+        output_desc: str = ""
+        return_desc: str = ""
+
+        def get_method_desc(short_desc: Optional[str], long_desc: Optional[str]) -> str:
+            """
+            method description is based on short desc and long desc, with 4 cases of outputs:
+            - if only short (or long) desc, then output only short (or long) desc
+            - if neither exists, then outputs empty string
+            - if both exists, concat with break line
+            """
+            if not short_desc:
+                return long_desc if long_desc else ""
+            else:
+                method_desc = short_desc
+                if long_desc:
+                    method_desc = method_desc + "\n" + long_desc
+                return method_desc
+
+        def get_ret_desc(_ret_desc: str, _output_desc: str) -> str:
+            """
+            return description is based on return desc and output desc:
+            - if only return (or output) desc, then output only return (or output) desc
+            - if neither exists, then outputs empty string
+            - if both exists, concat with space
+            """
+            if not _ret_desc:
+                return _output_desc
+            else:
+                res = _ret_desc
+                if _output_desc:
+                    res = res + " " + _output_desc
+                return res
+
+        if self.subroutine.implementation.__doc__ is not None:
```

Comment:
nit: I think the check on this line should simply 
```suggestion
        if self.subroutine.implementation.__doc__:
```
since we can't parse anything from an empty docstring either